From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Thu, 10 Oct 2019 23:30:16 +0200
Subject: Restore Search Ready Omnibox flag

Revert "Cleanup Search Ready Omnibox flag since it has launched"
This reverts commit ae458edcc8422d0815d0e82261e71fe10d7d6fc2.

Disable search-ready omnibox by default

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 chrome/browser/flags/android/chrome_feature_list.cc |  1 +
 .../chrome/browser/flags/ChromeFeatureList.java     |  1 +
 .../DropdownItemViewInfoListBuilder.java            |  8 +++++++-
 .../restore-Search-Ready-Omnibox-flag.inc           | 13 +++++++++++++
 .../restore-Search-Ready-Omnibox-flag.inc           |  3 +++
 .../restore-Search-Ready-Omnibox-flag.inc           |  1 +
 6 files changed, 26 insertions(+), 1 deletion(-)
 create mode 100644 cromite_flags/chrome/browser/about_flags_cc/restore-Search-Ready-Omnibox-flag.inc
 create mode 100644 cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/restore-Search-Ready-Omnibox-flag.inc
 create mode 100644 cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/restore-Search-Ready-Omnibox-flag.inc

diff --git a/chrome/browser/flags/android/chrome_feature_list.cc b/chrome/browser/flags/android/chrome_feature_list.cc
index e994f875df3563f172a5b4f7bfc227bf2b01e7f8..9c76495e3ef68b6ed63bb9e620718cec90de84e1 100644
--- a/chrome/browser/flags/android/chrome_feature_list.cc
+++ b/chrome/browser/flags/android/chrome_feature_list.cc
@@ -450,6 +450,7 @@ const base::Feature* const kFeaturesExposedToJava[] = {
     &kReadAloudIPHMenuButtonHighlightCCT,
     &kRecordSuppressionMetrics,
     &kReengagementNotification,
+    &kSearchReadyOmniboxFeature,
     &kRelatedSearchesAllLanguage,
     &kRelatedSearchesSwitch,
     &kReloadTabUiResourcesIfChanged,
diff --git a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
index 7e4cff971bb8911da390936bf52beaf782d28926..7df091b5a508cf813743a597e3cadf248c876a1c 100644
--- a/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
+++ b/chrome/browser/flags/android/java/src/org/chromium/chrome/browser/flags/ChromeFeatureList.java
@@ -627,6 +627,7 @@ public abstract class ChromeFeatureList {
     public static final String MIGRATE_MANAGEMENT_TO_WEBUI_ON_MOBILE =
             "MigrateManagementPageToWebUIOnMobile";
     public static final String MOST_VISITED_TILES_CUSTOMIZATION = "MostVisitedTilesCustomization";
+    public static final String SEARCH_READY_OMNIBOX = "SearchReadyOmnibox";
     public static final String MOST_VISITED_TILES_RESELECT = "MostVisitedTilesReselect";
     public static final String MOVE_TO_FRONT_IN_LAUNCH_INTENT_DISPATCHER =
             "MoveToFrontInLaunchIntentDispatcher";
diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java
index 79002abe70fe1954def545fb5436bc52d3a46709..df74118acc365cdfd7d54de92e214c857ee4f7d3 100644
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/suggestions/DropdownItemViewInfoListBuilder.java
@@ -14,6 +14,7 @@ import org.chromium.build.annotations.Initializer;
 import org.chromium.build.annotations.NullMarked;
 import org.chromium.build.annotations.Nullable;
 import org.chromium.chrome.browser.browser_controls.BrowserControlsStateProvider.ControlsPosition;
+import org.chromium.chrome.browser.flags.ChromeFeatureList;
 import org.chromium.chrome.browser.omnibox.UrlBarEditingTextStateProvider;
 import org.chromium.chrome.browser.omnibox.styles.OmniboxImageSupplier;
 import org.chromium.chrome.browser.omnibox.styles.OmniboxResourceProvider;
@@ -55,6 +56,7 @@ class DropdownItemViewInfoListBuilder {
     private @Nullable OmniboxImageSupplier mImageSupplier;
     private final BookmarkState mBookmarkState;
     private final MonotonicObservableSupplier<@ControlsPosition Integer> mToolbarPositionSupplier;
+    private EditUrlSuggestionProcessor mEditUrlSuggestionProcessor;
 
     DropdownItemViewInfoListBuilder(
             Supplier<@Nullable Tab> tabSupplier,
@@ -116,10 +118,11 @@ class DropdownItemViewInfoListBuilder {
         AutocompleteUIContext uiContext =
                 createUIContext(context, host, textProvider, actionDelegate);
 
-        registerSuggestionProcessor(new EditUrlSuggestionProcessor(uiContext));
+        mEditUrlSuggestionProcessor = new EditUrlSuggestionProcessor(uiContext);
         registerSuggestionProcessor(new AnswerSuggestionProcessor(uiContext));
         registerSuggestionProcessor(new ClipboardSuggestionProcessor(uiContext));
         registerSuggestionProcessor(new EntitySuggestionProcessor(uiContext));
+        registerSuggestionProcessor(mEditUrlSuggestionProcessor);
         registerSuggestionProcessor(new TailSuggestionProcessor(uiContext));
         registerSuggestionProcessor(new MostVisitedTilesProcessor(uiContext));
         registerSuggestionProcessor(new TabGroupSuggestionProcessor(uiContext));
@@ -185,6 +188,9 @@ class DropdownItemViewInfoListBuilder {
             mImageSupplier.onNativeInitialized();
         }
 
+        if (ChromeFeatureList.isEnabled(ChromeFeatureList.SEARCH_READY_OMNIBOX) == false) {
+            mPriorityOrderedSuggestionProcessors.remove(mEditUrlSuggestionProcessor);
+        }
         for (int index = 0; index < mPriorityOrderedSuggestionProcessors.size(); index++) {
             mPriorityOrderedSuggestionProcessors.get(index).onNativeInitialized();
         }
diff --git a/cromite_flags/chrome/browser/about_flags_cc/restore-Search-Ready-Omnibox-flag.inc b/cromite_flags/chrome/browser/about_flags_cc/restore-Search-Ready-Omnibox-flag.inc
new file mode 100644
index 0000000000000000000000000000000000000000..c4ba707074bfb482d8ad6deb1cf2335b0a46fbdb
--- /dev/null
+++ b/cromite_flags/chrome/browser/about_flags_cc/restore-Search-Ready-Omnibox-flag.inc
@@ -0,0 +1,13 @@
+#if BUILDFLAG(IS_ANDROID)
+
+#ifdef FLAG_SECTION
+
+    {"enable-search-ready-omnibox",
+     "Search Ready Omnibox",
+     "Clears the omnibox and adds a suggestion item to share, copy, or edit the "
+     "URL.", kOsAndroid,
+     FEATURE_VALUE_TYPE(chrome::android::kSearchReadyOmniboxFeature)},
+
+#endif
+
+#endif
diff --git a/cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/restore-Search-Ready-Omnibox-flag.inc b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/restore-Search-Ready-Omnibox-flag.inc
new file mode 100644
index 0000000000000000000000000000000000000000..86b20add01b7fbd013024f001e67af7cd9cb12c4
--- /dev/null
+++ b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_cc/restore-Search-Ready-Omnibox-flag.inc
@@ -0,0 +1,3 @@
+CROMITE_FEATURE(kSearchReadyOmniboxFeature,
+               "SearchReadyOmnibox",
+               base::FEATURE_ENABLED_BY_DEFAULT);
diff --git a/cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/restore-Search-Ready-Omnibox-flag.inc b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/restore-Search-Ready-Omnibox-flag.inc
new file mode 100644
index 0000000000000000000000000000000000000000..14d4e53eeacd1f2a7945f4a0578a30250d6cbec1
--- /dev/null
+++ b/cromite_flags/chrome/browser/flags/android/chrome_feature_list_h/restore-Search-Ready-Omnibox-flag.inc
@@ -0,0 +1 @@
+BASE_DECLARE_FEATURE(kSearchReadyOmniboxFeature);
